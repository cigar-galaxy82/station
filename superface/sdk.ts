import { createTypedClient } from "@superfaceai/one-sdk";

import { communicationSendEmail } from "./types/communication/send-email";
import { communicationSendMessage } from './types/communication/send-message';

export { CommunicationSendEmailProfile } from "./types/communication/send-email";
export const typeDefinitions = {
    ...communicationSendEmail,
    ...communicationSendMessage
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
