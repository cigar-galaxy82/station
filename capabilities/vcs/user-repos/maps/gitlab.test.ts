import { ok, SuperfaceClient } from '@superfaceai/sdk';

describe('vcs/user-repos/gitlab', () => {
    it('performs correctly', async () => {
        const client = new SuperfaceClient;
        const profile = await client.getProfile('vcs/user-repos');
        const useCase = profile.getUseCase('userRepos');
        const provider = await client.getProvider('gitlab');
        await expect(
            useCase.perform({ user: 'zdne' }, { provider })
        ).resolves.toEqual(
            ok({
                repos: [{ name: 'test', description: 'Hello World!' }]
            })
        );
    })
})
