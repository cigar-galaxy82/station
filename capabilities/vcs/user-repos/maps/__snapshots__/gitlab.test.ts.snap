// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/user-repos/gitlab UserRepos should map eror successfully 1`] = `
Object {
  "error": "MappedError: Expected error
Properites: {
  \\"message\\": \\"404 User Not Found\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[0].statements[0]
Original Map Location: Line 11, column 5",
}
`;

exports[`vcs/user-repos/gitlab UserRepos should perform successfully 1`] = `
Object {
  "value": Object {
    "repos": Array [
      Object {
        "description": "Hello World!",
        "name": "test",
      },
    ],
  },
}
`;

exports[`vcs/user-repos/gitlab UserRepos should perform successfully on repositories with empty descriptions 1`] = `
Object {
  "value": Object {
    "repos": Array [
      Object {
        "description": undefined,
        "name": "empty-test",
      },
      Object {
        "description": undefined,
        "name": "ChatApp",
      },
    ],
  },
}
`;
